lib/datadog/core/configuration/components.rb

```diff
@@ -81,13 +82,17 @@ def build_telemetry(settings, agent_settings, logger)
         def initialize(settings)
           @logger = self.class.build_logger(settings)
 
+          # This agent_settings is intended for use within Core. If you require
+          # agent_settings within a product outside of core you should extend
+          # the Core resolver from within your product/component's namespace.
           agent_settings = AgentSettingsResolver.call(settings, logger: @logger)
 
           @remote = Remote::Component.build(settings, agent_settings)
-          @tracer = self.class.build_tracer(settings, agent_settings)
+          @tracer = self.class.build_tracer(settings, logger: @logger)
```

Comment:
Why do we pass `@logger`? Is this because it might need to log during initialization?

Reply:
Correct. It gets used by the AgentSettingsResolver initializer, which now gets done in the specific component initializers, rather than here.